frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,223 @@
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared';
+import {
+  TableComposable,
+  TableGridBreakpoint,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+
+import { ResourceLink } from './utils';
+import { createRequests } from './edit-yaml';
+import { K8sKind } from '../module/k8s';
+
+/**
+ * Without this prop our current TS types fail to match and require a `translate` prop to be added. PF suggests we
+ * update our types, but that causes other issues. This will have to do as a workaround for now.
+ *
+ * This is the best that I can find relates to this value:
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3423b4fc3e3da09f8acc386bc2fee6fb8f5e0880/types/react/index.d.ts#L1763
+ */
+const reactPropFix = {
+  translate: 'no',
+};
+
+const ImportYAMLPageStatus: React.FC<ImportYAMLStatusProps> = ({ errors, sending }) => {
+  const { t } = useTranslation();
+  let StatusBlock;
+
+  if (sending) {
+    StatusBlock = (
+      <>
+        <Spinner size="lg" />
+        <h2>{t('public~Creating resources...')}</h2>
+      </>
+    );
+  } else if (!sending && !errors) {
+    StatusBlock = (
+      <>
+        <GreenCheckCircleIcon size="lg" />
+        <h2>{t('public~Resources successfully created')}</h2>
+      </>
+    );
+  } else {
+    StatusBlock = (
+      <>
+        <YellowExclamationTriangleIcon size="lg" />
+        <h2>{t('public~One or more resources failed to be created')}</h2>
+      </>
+    );
+  }
+  return <div className="co-import-yaml-status text-center">{StatusBlock}</div>;
+};
+
+const ImportYAMLResourceStatus: React.FC<ImportYAMLResourceStatusProps> = ({ status }) => {
+  let StatusIcon;
+  if (status.creating) {
+    StatusIcon = <Spinner size="sm" className="co-icon-space-r" />;
+  } else if (status.error) {
+    StatusIcon = <RedExclamationCircleIcon className="co-icon-space-r" size="sm" />;
+  } else {
+    StatusIcon = <GreenCheckCircleIcon className="co-icon-space-r" size="sm" />;
+  }
+  return (
+    <span>
+      {StatusIcon}
+      {status.message}
+    </span>
+  );
+};
+
+export const ImportYAMLResults: React.FC<ImportYAMLResultsProps> = ({
+  importResources,
+  models,
+  retryFailed,
+  showResults,
+}) => {
+  const { t } = useTranslation();
+  const [resources, setResources] = React.useState(
+    importResources.map((obj) => ({
+      obj: { ...obj },
+      status: { creating: true, message: t('public~Creating') },
+    })),
+  );
+  const [inFlight, setInFlight] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  React.useEffect(() => {
+    const requests = createRequests(importResources, models);
+    Promise.allSettled(requests).then(() => {
+      setInFlight(false);
+    });
+    requests.forEach((resourceRequest, index) => {
+      resourceRequest
+        .then((resourceObj) =>
+          setResources((prevResources) => [
+            ...prevResources.slice(0, index),
```

Comment:
I would just copy the array and set the one value in the new array.

```ts
const nextResources = [...prevResources];
nextResources[index] = { ... };
return nextResources;
```

Reply:
yes this is a bit simpler. thanks for the tip